api/src/services/HierarchyCollector.ts

```diff
@@ -25,21 +25,19 @@ class HierarchyCollector {
         return metadata;
     }
 
-    protected extractRelations(RELS: string): {[key: string]: Array<string>} {
-        let xmlParser = new DOMParser();
-        let RELS_XML = xmlParser.parseFromString(RELS, "text/xml");
-        let rdfXPath = xpath.useNamespaces({
+    protected extractRelations(RELS: string): Record<string, Array<string>> {
+        const xmlParser = new DOMParser();
+        const RELS_XML = xmlParser.parseFromString(RELS, "text/xml");
+        const rdfXPath = xpath.useNamespaces({
             rdf: "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
         });
-        let relations: {[key: string]: Array<string>} = {};
-        rdfXPath(
-            '//rdf:Description/*', RELS_XML
-        ).forEach((relation) => {
-            let values = rdfXPath('text()', relation);
+        const relations: Record<string, Array<string>> = {};
+        rdfXPath("//rdf:Description/*", RELS_XML).forEach((relation: Node) => {
+            let values = rdfXPath("text()", relation) as Array<Node>;
             // If there's a namespace on the node name, strip it:
-            let nodeName = relation.nodeName.split(':').pop();
+            const nodeName = relation.nodeName.split(":").pop();
             if (values.length === 0) {
-                values = rdfXPath('./@rdf:resource', relation);
+                values = rdfXPath("./@rdf:resource", relation) as Array<Node>;
```

Comment:
This is an important convention I didn't know about if you need to specify which overloaded type gets returned.